topi/python/topi/cuda/nms.py

```diff
@@ -5,427 +5,66 @@
-        if i < axis:
-            axis_mul_before *= data.shape[i]
-        elif i > axis:
-            axis_mul_after *= data.shape[i]
-
-    dshape = axis_mul_before*axis_mul_after
-    fshape = data.shape[axis] * dshape
-
-    loc_buf = api.decl_buffer(dshape, index.dtype, "sizes", data_alignment=8)
-    new_index_buf = api.decl_buffer(
-        fshape, index.dtype, "index_new", data_alignment=8)
-    out_index_buf = api.decl_buffer(
-        fshape, index.dtype, "index_out", data_alignment=8)
-    new_data_buf = api.decl_buffer(
-        dshape, data.dtype, "data_new", data_alignment=8)
-
-    loc = \
-        tvm.extern([(dshape,)],
-                   [index],
-                   lambda ins, outs: sort_pre_ir(
-                       ins[0], outs[0], axis_mul_before, axis_mul_after),
-                   dtype=[index.dtype],
-                   in_buffers=index_buf,
-                   out_buffers=[loc_buf],
-                   tag="sorting_prepare")
-
-    data_new, index_new = \
-        tvm.extern([(dshape,), (fshape,)],
-                   [data, index, loc],
-                   lambda ins, outs: sort_pre_ir_data(
-                       ins[0], ins[1], ins[2], outs[0], outs[1], axis,
-                       axis_mul_before, axis_mul_after),
-                   dtype=[data.dtype, index.dtype],
-                   in_buffers=[data_buf, index_buf, loc_buf],
-                   out_buffers=[new_data_buf, new_index_buf],
-                   tag="sorting_data")
-
-    index_out = \
-        tvm.extern([(fshape,)],
-                   [data, index, data_new, index_new, loc],
-                   lambda ins, outs: sort_oet_ir(
-                       ins[0], ins[1], ins[2], ins[3], ins[4], outs[0],
-                       axis_mul_before, axis_mul_after, axis, is_descend),
-                   dtype=[index.dtype],
-                   in_buffers=[data_buf, index_buf,
-                               new_data_buf, new_index_buf, loc_buf],
-                   out_buffers=[out_index_buf],
-                   tag="sorting_oet")
-    out = \
-        tvm.extern([data.shape],
-                   [data, index, index_out, loc],
-                   lambda ins, outs: sort_ir_out(
-                       ins[0], ins[1], ins[2], ins[3], outs[0],
-                       axis_mul_before, axis_mul_after, axis),
-                   dtype=[index.dtype],
-                   in_buffers=[data_buf, index_buf, out_index_buf, loc_buf],
-                   out_buffers=output_buf,
-                   tag="sorting_output")
-    return out
-
+    ib.scope_attr(bx, "virtual_thread", nthread_bx)
+    tid = bx * nthread_tx + tx
+    temp_data = ib.allocate("float32", (1,), name="temp_data", scope="local")
+    temp_index = ib.allocate("int32", (1,), name="temp_index", scope="local")
+
+    with ib.for_range(0, batch, for_type="unroll") as b:
+        start = b * num_anchors
+        with ib.if_scope(tid < num_anchors):
+            p_out[start + tid] = tid
```

Comment:
seems storage_sync is missing here, I will update my pr

Reply:
@vinx13 Would you like to seperate argsort to a seperate file so that we can share the use of it? I can add unitest to it if needed.

Reply:
Maybe we can make argsort a normal topi op? I'll add cpu implementation later.

Reply:
@vinx13 p_index is the valid_count variable which is a 1D array resulted from the multibox operators.

Reply:
@Laurawly shouldn't be p_index[batch_id]? are you assuming batch = 1?

Reply:
@kevinthesun @Laurawly The difficulty of sharing argsort (or extract it as a topi operator) is that we hope `sort_num` can be either a `tvm.Tensor` or constant array, but we can't use tvm.Expr to subscript a python array. Do you have ideas?